File: cpp/questions/maxMatrixSum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int mn = INT_MAX;
        long long ans = 0;
        int neg_cnt = 0;
        for (auto& row : matrix) {
            for (auto& num : row) {
                if (num <= 0) {
                    ans += -num;
                    mn = std::min(mn, -num);
                    neg_cnt += 1;
                } else {
                    ans += num;
                    mn = std::min(mn, num);
                }
            }
        }
        if (neg_cnt%2 == 0) {
            return ans;
        }
        return ans-mn*2;
    }
};

File: python/questions/maxMatrixSum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        ans = 0
        neg_cnt = 0
        mn = inf
        for r in range(len(matrix)):
            for c in range(len(matrix[0])):
                if matrix[r][c] < 0:
                    neg_cnt += 1
                    ans += -matrix[r][c]
                    mn = min(mn, -matrix[r][c])
                else:
                    ans += matrix[r][c]
                    mn = min(mn, matrix[r][c])
        if neg_cnt%2 == 0:
            return ans
        return ans-mn*2
